### PR TITLE
dsg-1341: handle users state initialization

### PR DESCRIPTION
**Pull request by davidyoung77**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

- Load initial users state only on authentication
- Fix allowed node engine for nx console
- Create app state library
- Move user state to app state library
- rename user routes service to match our others

**Note:** Most of the changes here are related to renaming creating a new library and moving and renaming the users services, I have added comments to the files that are directly related to code updates

## Screenshot of the affected changes

Screenshot 2023-09-19 at 9 01 43 AM
Initialize user state request
Screenshot 2023-09-19 at 9 02 33 AM

New app-state library

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/agency/shell/src/lib/components/shell/shell.component.ts [hunk 1]
```diff
@@ -64,7 +67,14 @@ export class ShellComponent implements OnInit {
   ) {}
 
   public ngOnInit() {
-    this._userStateService.getAgencyUsers$().subscribe();
+    this._authenticationService.isAuthenticated$
```
Comment (davidyoung77) [inline libs/agency/shell/src/lib/components/shell/shell.component.ts:70]: We initialize the user state only on autentication

File: libs/shared/feature/app-state/src/lib/services/user-state/user-state.service.spec.ts
Comment (davidyoung77) [inline libs/shared/feature/app-state/src/lib/services/user-state/user-state.service.spec.ts:1]: Adding a lot of test coverage here

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.0",
   "license": "MIT",
   "engines": {
-    "node": "^14.20.0 || ^16.13.0 || ^18.10.0",
+    "node": "^14.20.0 || ^16.13.0 || ^18.10.0 || ^20.5.0",
```
Comment (davidyoung77) [inline package.json:6]: Accept node version for nx console

**Review by Ortega-Dan — APPROVED** [review]

**Review by odeljanin — APPROVED** [review]
LGTM

**Pull request merged**